### PR TITLE
WIP: Check if chart needs auth

### PR DESCRIPTION
**Pull request by mrajashree**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 1]
```diff
@@ -137,6 +138,24 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 	}
 	defer cleanup(ctx, deps, &err)
 
+	if clusterSpec.VersionsBundle.Cilium.HelmChart.Image() != "public.ecr.aws/isovalent/cilium" {
+		if clusterConfig.Spec.RegistryMirrorConfiguration == nil {
+			return fmt.Errorf("provide registry mirror configuration to pull cilium chart from")
+		}
+		chart := clusterSpec.VersionsBundle.Cilium.HelmChart
+		if err := deps.Helm.PullChart(ctx, fmt.Sprintf("%s%s", ociPrefix, chart.Image()), chart.Tag()); err != nil {
```
Comment (jaxesn) [inline cmd/eksctl-anywhere/cmd/createcluster.go:146]: I'm wondering if we really support local registries with auth tho? Wouldn't we have issues pulling images during creation if the registry has auth?

Comment (mrajashree) [reply cmd/eksctl-anywhere/cmd/createcluster.go:146]: if we find that we don't need this auth, we'll have to remove [this](https://github.com/aws/eks-anywhere/blob/main/cmd/eksctl-anywhere/cmd/importimages.go#L58) from import-images too

Comment (vivek-koppuru) [reply cmd/eksctl-anywhere/cmd/createcluster.go:146]: It's because the repo we were testing was a private one, made it public and it works now

**Comment by mrajashree** [comment]
closing as not needed

**Pull request closed without merging**